### PR TITLE
S3 Benchmarking Tool

### PR DESCRIPTION
**Pull request by taburet**
See src/s3-bench/README.md for details.

### CODE REVIEW

File: src/s3_bench/main.rs
Comment (criccomini) [inline src/s3_bench/main.rs:47]: I think we might want the last write to be durable to guarantee that all writes are flushed before we start running the actual benchmark. Otherwise, we might be measuring the tail end of the preload. Not required for this PR, but wanted to make a note of it.

File: src/s3_bench/main.rs
Comment (criccomini) [inline src/s3_bench/main.rs:134]: Should this be using `root_folder` since the object cache path is configurable?

**Review by criccomini — APPROVED** [review]
LGTM. Couple minor notes, but nothing blocking. I'll let it sit for 24h then merge if nothing changes.

File: src/s3_bench/main.rs
Comment (taburet) [reply src/s3_bench/main.rs:47]: could the same be achieved with `db.flush()`?

File: src/s3_bench/main.rs
Comment (taburet) [reply src/s3_bench/main.rs]: fixed in https://github.com/slatedb/slatedb/pull/233/commits/7bfa9e90974bb4d588f9dbe4ea98fae1679b4aff

File: src/s3_bench/main.rs
Comment (taburet) [reply src/s3_bench/main.rs:51]: have tried it, fails with `Fenced` status, reverted to a write with `await_durable` set to `true`: https://github.com/slatedb/slatedb/pull/233/commits/33c74cc56ed5158261ec819d25d8e0d3c5595054

Comment (criccomini) [reply src/s3_bench/main.rs:51]: 🤔 That sounds like a bug. Flush should work. I'll open a separate GH issue for that.

Comment (criccomini) [reply src/s3_bench/main.rs:51]: Opened https://github.com/slatedb/slatedb/issues/235

**Review by criccomini — APPROVED** [review]

**Comment by rodesai** [comment]
Should we do some follow-on work to combine this with db_bench? Seems like we have a couple tools now which do similar things

**Comment by criccomini** [comment]
> Should we do some follow-on work to combine this with db_bench? Seems like we have a couple tools now which do similar things 

Yea, I think so. I didn't want to block on this tho. I'm planning to take up the nightly build so I don't mind taking some of the other refactoring. Here's some thoughts:

1. Switch print to info (except for gnu plot output).
2. Move to Admin commands
3. Share S3 setup logic with db bench.
4. Look at combining benchmark logic with db bench (I haven't dug into this yet).

What else am I missing?

**Comment by criccomini** [comment]
Opened https://github.com/slatedb/slatedb/issues/236

**Pull request merged**
